### ISSUE TITLE
Re-implement TF-IDF in Normalize options

### ISSUE THREAD

**Issue by scottkleinman** [WheatonCS/Lexos#444]
We need a fresh implementation of TF-IDF weighting. See issue #437 for the background.

**Comment by chantisnake** [comment]
no body really understand how TF-IDF works, that is the problem...

**Comment by scottkleinman** [comment]
OK, I'll look into it as soon as I can.

**Comment by scottkleinman** [comment]
[This](http://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.TfidfTransformer.html#sklearn.feature_extraction.text.TfidfTransformer) helps a lot. Simply using `tf * idf` as the formula will match the standard accounts known to humanists. This would be equivalent to the `norm: None` setting. We can find out if there is any demand for fine-tuning the normalisation parameters at a later date. The real question is whether we should offer (or even impose) the `smooth_idf` setting (which is the default). This adds 1 to to document frequencies, "as if an extra document was seen containing every term in the collection exactly once" (I'm quoting the sklearn manual here.) "The effect of this is that terms with zero idf, i.e. that occur in all documents of a training set, will not be entirely ignored." If we can decide whether this refinement is worthwhile for Lexos, the implementation should be relatively easy.

**Comment by scottkleinman** [comment]
I just did some testing using both `sklearn.feature_extraction.text.TfidfTransformer` and `sklearn.feature_extraction.text.TfidfVectorizer`. If I set `Norm=None` I get a matrix exactly the same as the raw counts. If I use any other normalisation setting, I get an empty matrix. We may have to play with these functions outside of Lexos to figure out where the problem is.

In the meantime, I changed the commented out html in `base_analyze.html` a little bit to make it easier to test when we get around to it.

**Comment by scottkleinman** [comment]
After discussion, we've decided to leave this for Lexos 3.5 and label this issue an enhancement.
